### ISSUE TITLE
Help: DNSOMatic Error Malformed Password

### ISSUE THREAD

**Issue by muslimsteel** [qdm12/ddns-updater#448]
**TLDR**: *Describe your issue in a one liner here*
Hello, I am trying to get DNSOMatic to update using DDNSUpdater, but for whatever reason, it is giving me ERROR malformed password. I tried turning on debug logging but all that did was add the config file to the logs and then it shuts down. Am I missing something? My password is 20 characters long and only has one special character in it a "$" sign.
I posted this originally in Discussions on the GitHub page but have not got a response there so I thought I might try here. 
Thanks for the help!

1. Is this urgent: No
2. DNS provider(s) you use: DNSOMatic
3. Program version:

    
`Running version latest built on 2023-02-02T00:41:38Z (commit 228fc50)`

4. What are you using to run the container: docker under UnRaid
5. Extra information (optional)

Logs:
```
========================================
========================================
============= ddns-updater =============
========================================
=========== Made with ❤️ by ============
======= https://github.com/qdm12 =======
========================================
========================================

Running version latest built on 2023-02-02T00:41:38Z (commit 228fc50)

🔧 Need help? https://github.com/qdm12/ddns-updater/discussions/new
🐛 Bug? https://github.com/qdm12/ddns-updater/issues/new
✨ New feature? https://github.com/qdm12/ddns-updater/issues/new
☕ Discussion? https://github.com/qdm12/ddns-updater/discussions/new
💻 Email? quentin.mcgaw@gmail.com
💰 Help me? https://www.paypal.me/qmcgaw https://github.com/sponsors/qdm12
2023-03-14T15:54:43-04:00 INFO reading JSON config from file /updater/data/config.json
2023-03-14T15:54:43-04:00 DEBUG config read: {
  "settings": [
    {
      "provider": "dnsomatic",
      "domain": "*",
      "host": "*",
      "username": "xxxxxxxxxxxxxx@gmail.com",
      "password": "xxxxxxxxxxxxxxxxxxxx",
      "ip_version": "ipv4"
    }
  ]
}
2023-03-14T15:54:43-04:00 ERROR malformed password
2023-03-14T15:54:43-04:00 INFO Shutdown successful```

Configuration file (**remove your credentials!**):

```see above

```

Host OS:
Unraid 6.11.5

**Comment by beneliason** [comment]
I'm experiencing the same issue (unraid; docker). When I add a simple (but incorrect) password `ddns-updater` will start with bad auth. OpenDNS requires a password with special characters. (seems to be a bug) 🤷🏻‍♂️

**Comment by mrt2nbl** [comment]
Same here :/

**Comment by qdm12** [comment]
First sorry for the massive delay;

From the documentation https://www.dnsomatic.com/docs/api :

> DNS-O-Matic passwords are 6-20 characters. All ASCII letters and numbers accepted. Dots, dashes, and underscores allowed, but not at the beginning or end of the string.

Hence the existing regex to enforce this `^[a-zA-Z0-9][a-zA-Z0-9._-]{5,19}$`, should symbols be accepted?? 🤔

**Comment by mrt2nbl** [comment]
I tried that changes on my own fork and I did the trick.
dnsOMaticPassword = regexp.MustCompile(^[a-zA-Z0-9 !@#$€%&+*._-]{5,19}$)
DNS-O-Matic accept special characters

**Comment by muslimsteel** [comment]
Thank you!
